src/pilot/assign_role.py

```diff
@@ -406,8 +419,17 @@ def define_storage_logical_disks(drac_client, raid_controller_name):
             "configuration")
         return None
 
-    os_logical_disk = define_storage_operating_system_logical_disk(
-        raid_cntlr_physical_disks, drac_client, raid_controller_name)
+    if len(raid_controllers) > 1:
+        boss_controller = [cntrl for cntrl in raid_controllers
+                           if drac_client.is_boss_controller(cntrl)][0]
+        if boss_controller:
+            os_logical_disk = define_storage_operating_system_logical_disk(
+                raid_cntlr_physical_disks[boss_controller],
+                drac_client, boss_controller)
```

Comment:
let's add an else clause here and pick the first RAID controller in the raid_controllers list that has 2 or more drives, then call into define_storage_operating_system_logical_disk() passing that controller.  This would handle the case of having multiple controllers where neither is a BOSS controller.  I'm honestly not sure that this would be a valid configuration, but I would rather our code do something reasonable in this case just to be sure.  If there are no RAID controllers with 2 or more drives, then we should log the error on line 412 above and return None.

Reply:
Sure Chris. I have handled the code for multiple raid controllers. And tested those changes with existing controller configuration. It is working fine.